cpp/src/arrow/CMakeLists.txt

```diff
@@ -24,6 +24,8 @@ install(FILES
   schema.h
   table.h
   type.h
+  type_fwd.h
```

Comment:
I'm not sure there's a hard and fast rule. Kudu seems to use dashes for internal files, and `_` as a separator in "public" headers:

https://github.com/apache/kudu/tree/master/src/kudu/client

so you would have

`widget_xyz-test.cc`

or

`widget_xyz-internal.h`

https://google.github.io/styleguide/cppguide.html#File_Names

let me know what you think